src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -90,94 +92,68 @@ int IVsEditorFactory.Close()
-
-                    try
-                    {
-                        designerLoader.Initialize(_oleServiceProvider, vsHierarchy, (int)itemid, (IVsTextLines)textBuffer);
-                        pbstrEditorCaption = designerLoader.GetEditorCaption((int)readOnlyStatus);
-
-                        var designer = designerService.CreateDesigner(_oleServiceProvider, designerLoader);
-                        ppunkDocView = Marshal.GetIUnknownForObject(designer.View);
-                        pguidCmdUI = designer.CommandGuid;
-                    }
-                    catch
-                    {
-                        designerLoader.Dispose();
-                        throw;
-                    }
+            ppunkDocData = Marshal.GetIUnknownForObject(textBuffer);
 
-                    break;
+            return VSConstants.S_OK;
+        }
+
+        public object GetDocumentData(uint grfCreate, string pszMkDocument, IVsHierarchy pHier, uint itemid)
+        {
+            var editorAdaptersFactoryService = _componentModel.GetService<IVsEditorAdaptersFactoryService>();
 
-                case "Code":
+            var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
+            var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
+            IVsTextBuffer? textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
 
-                    var codeWindow = editorAdaptersFactoryService.CreateVsCodeWindowAdapter(_oleServiceProvider);
-                    codeWindow.SetBuffer((IVsTextLines)textBuffer);
+            if (_encoding)
+            {
+                if (textBuffer is IVsUserData userData)
+                {
+                    // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
+                    var hresult = userData.SetData(
+                        VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
+                        (uint)__PROMPTONLOADFLAGS.codepagePrompt);
 
-                    codeWindow.GetEditorCaption(readOnlyStatus, out pbstrEditorCaption);
+                    Marshal.ThrowExceptionForHR(hresult);
+                }
+            }
 
-                    ppunkDocView = Marshal.GetIUnknownForObject(codeWindow);
-                    pguidCmdUI = VSConstants.GUID_TextEditorFactory;
+            return textBuffer;
+        }
 
-                    break;
+        public object GetDocumentView(uint grfCreate, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, uint itemid)
+        {
+            throw new NotImplementedException();
+        }
 
-                default:
+        public string GetEditorCaption(string pszMkDocument, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, out Guid pguidCmdUI)
+        {
+            Contract.ThrowIfFalse(punkDocData != IntPtr.Zero, $"Document data is null for {pszMkDocument}");
```

Comment:
this is an API we're exposing right?  if the data is wrong, we shoudl throw an ArugmentException.